app/controllers/kiosk_reservations_controller.rb

```diff
@@ -102,7 +102,11 @@ def respond_error(message)
   def check_kiosk_enabled
     return if current_facility&.kiosk_enabled? && SettingsHelper.feature_on?(:kiosk_view)
 
-    raise NUCore::PermissionDenied
+    if current_user
+      render "errors/forbidden", status: :forbidden
+    else
+      redirect_to new_user_session_path, alert: "You need to login to access this page." and return
```

Comment:
Why do we need to `return`? Is it the same as doing this?

```suggestion
      return redirect_to new_user_session_path, alert: "You need to login to access this page."
```